File: app/src/main/java/at/droelf/travellogapp/NetworkChangeListener.java
package at.droelf.travellogapp;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

public class NetworkChangeListener extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if(intent.getExtras() != null) {
            final ConnectivityManager systemService = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            final NetworkInfo activeNetworkInfo = systemService.getActiveNetworkInfo();

            final boolean online = activeNetworkInfo != null && activeNetworkInfo.isConnectedOrConnecting();

            Log.d(getClass().getSimpleName(), "Network status changed, online: " + online);

            if(online) {
                context.startService(new Intent(context.getApplicationContext(), ImageUploadAndroidService.class));
            }
        }
    }
}

File: app/src/main/scala/at/droelf/travellogapp/NetworkChangeListener.scala
package at.droelf.travellogapp

import android.content.{Intent, Context, BroadcastReceiver}
import android.net.{NetworkInfo, ConnectivityManager}
import android.util.Log

class NetworkChangeListener extends BroadcastReceiver {
  def onReceive(context: Context, intent: Intent) {
    if (intent.getExtras != null) {
      val systemService: ConnectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE).asInstanceOf[ConnectivityManager]
      val activeNetworkInfo: NetworkInfo = systemService.getActiveNetworkInfo
      val online: Boolean = activeNetworkInfo != null && activeNetworkInfo.isConnectedOrConnecting
      Log.d(getClass.getSimpleName, "Network status changed, online: " + online)
      if (online) {
        context.startService(new Intent(context.getApplicationContext, classOf[ImageUploadAndroidService]))
      }
    }
  }
}